File: 240124/잔해물을 덮기 위한 사각형의 최소 넓이/minimum-area-of-rectangle-to-cover-debris.py
a,b,c,d = map(int,input().split())
a1,b1,c1,d1 = map(int,input().split())
tal = (c-a) * (d-b)
square = [[0]*2001 for _ in range(2001)]
a+=1000
b+=1000
c+=1000
d+=1000

a1+=1000
b1+=1000
c1+=1000
d1+=1000
for i in range(a,c):
    for j in range(b,d):
        square[i][j] = 1

for j in range(a1,c1):
    for i in range(b1,d1):
        square[j][i] = 2

min_x ,max_x ,min_y , max_y = 20001,0,20001,0
exist = False

for i in range(2001):
    for j in range(2001):

        if square[i][j] == 1:
            exist = True

            min_x = min(min_x,i)
            max_x = max(max_x,i)
            min_y = min(min_y,j)
            max_y = max(max_y,j)

if exist:
    area = (max_x-min_x+1) * (max_y - min_y+1)
else:
    area = 0
print(area)

File: 241204/잔해물을 덮기 위한 사각형의 최소 넓이/minimum-area-of-rectangle-to-cover-debris.java
import java.util.*;
public class Main {
    public static int[][] paper = new int[2001][2001];
    public static int size,total,max_x,max_y,min_x,min_y;
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        
        int now = 1000;
        for(int s = 0; s<2; s++){
            int x1 = sc.nextInt();
            int y1 = sc.nextInt();
            int x2 = sc.nextInt();
            int y2 = sc.nextInt();
            for(int i = x1+now; i<x2+now; i++){
                for(int j = y1 + now; j<y2+now; j++){
                    if(s == 0){
                        paper[i][j] = 1;
                    }else{
                        paper[i][j] = 2 ;
                    }
                }
            }
        }
        min_x = 1000001;
        min_y = 1000001;
        boolean square = false;
        for(int i = 0; i<2001; i++){
            for(int j = 0; j<2001; j++){
                if(paper[i][j] == 1){
                    square = true;
                    min_x = Math.min(min_x,i);
                    max_x = Math.max(max_x,i);
                    min_y = Math.min(min_y,j);
                    max_y = Math.max(max_y,j);
                }
            }
        }
        if(!square){
            System.out.println(0);
        }else{
            System.out.println((max_x-min_x+1)*(max_y-min_y+1));
        }
        
        
        

    }
}